backend/de.metas.business/src/main/java-legacy/org/compiere/model/MPayment.java

```diff
@@ -342,9 +342,9 @@ && getC_BPartner_ID() != 0
 			if (getC_Order_ID() != 0)
 			{
 				final I_C_Order ord = getC_Order();
-				Check.errorIf(ord.getC_BPartner_ID() != getC_BPartner_ID(),
-						"Payment {} has C_BPartner_ID={}, but order {} has C_BPartner_ID={}",
-						this, getC_BPartner_ID(), ord, ord.getC_BPartner_ID());
+				Check.errorIf(ord.getBill_BPartner_ID() != getC_BPartner_ID(),
```

Comment:
pls introduce a proper AD_Message (with params) and use it.
we must use the opportunity to make this error user friendly.

Important: the mesage shall tell the user what's the problem and how he/she can fix it